DSpace/DSpace
# Authorization for Downloads of restricted Bitstreams: Short lived token endpoint

## References
_Add references/links to any related tickets or PRs. These may include:_
* Link to [REST Contract](https://github.com/DSpace/Rest7Contract/pull/126)

## Description
This PR adds support for short lived token generation. The JWTTokenHandler class used to generate our session tokens has been split up into 2 parts:
* SessionJWTTokenHandler
* ShortLivedJWTTokenHandler

The difference being when that they have different configuration properties. If you have existing configuration that differs from the default ones you will need to look at the authentication.cfg file changes to see the new properties. Furthermore the "expiration" configuration property has been changed to use milliseconds instead of minutes.

## Instructions for Reviewers
If you want to test this PR locally alter the _jwt.shortLived.token.expiration_ in the local.cfg & put it to something like 30 seconds (as 2 seconds is very short if you want to test this).

To test this pr:
* Request a short lived token
* Provide the token to a private bitstream using the "token" parameter
* Wait until the configured time expires & try to download the file again, it should fail now

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [X] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/master/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
